### PR TITLE
fix(terraform): Update to new AWS SDK

### PR DESCRIPTION
**Pull request by candiduslynx**
Inspired by #4611

### CODE REVIEW

**Review by erezrokah — APPROVED** [review]
I changed this to `fix` so it triggers a release

File: plugins/source/k8s/go.mod [hunk 1]
```diff
@@ -5,6 +5,7 @@ go 1.19
 require (
 	github.com/cloudquery/plugin-sdk v1.5.2
 	github.com/golang/mock v1.6.0
+	github.com/google/gnostic v0.5.7-v3refs
```
Comment (erezrokah) [inline plugins/source/k8s/go.mod:8]: This k8s change seems unrelated to this PR, can we do it in another one?

**Review by erezrokah — CHANGES_REQUESTED** [review]
Thanks @candiduslynx, I think we should move the k8s change to another PR, WDYT?

File: plugins/source/terraform/client/backend.go [hunk 2]
```diff
@@ -61,53 +62,49 @@ func parseAndValidate(reader io.Reader) (*TerraformData, error) {
...
 
-func NewS3TerraformBackend(config *BackendConfigBlock) (*TerraformBackend, error) {
+func NewS3TerraformBackend(ctx context.Context, cfg *BackendConfigBlock) (*TerraformBackend, error) {
 	var b S3BackendConfig
-	if err := json.Unmarshal(*config.Config, &b); err != nil {
+	if err := json.Unmarshal(*cfg.Config, &b); err != nil {
 		return nil, fmt.Errorf("cannot parse s3 backend config: %w", err)
 	}
 
+	awsConfig, err := config.LoadDefaultConfig(context.TODO())
```
Comment (disq) [inline plugins/source/terraform/client/backend.go:71]: Not context.TODO :)
```suggestion
	awsConfig, err := config.LoadDefaultConfig(ctx)
```

**Review by disq — CHANGES_REQUESTED** [review]
Single issue with a rogue context.TODO()

File: plugins/source/k8s/go.mod (hunk 1 above)
Comment (candiduslynx) [reply plugins/source/k8s/go.mod:8]: we can, I just ran `go mod tidy` in all of our plugins

Comment (erezrokah) [reply plugins/source/k8s/go.mod]: Got it, but if we merge this PR a new changelog entry for k8s will be created with `Update to new AWS SDK` which is a bit confusing

Comment (erezrokah) [reply plugins/source/k8s/go.mod]: I see that you reverted it, thanks!

**Review by erezrokah — APPROVED** [review]

**Review by disq — APPROVED** [review]
